lib/engine/game/g_18_zoo.rb

```diff
@@ -0,0 +1,631 @@
+# frozen_string_literal: true
+
+require_relative '../config/game/g_18_zoo'
+require_relative 'base'
+
+module Engine
+  module Game
+    class G18ZOO < Base
+      load_from_json(Config::Game::G18ZOO::JSON_MAP_A, Config::Game::G18ZOO::JSON_CERT_LIMIT_MAP_SMALL,
+                     Config::Game::G18ZOO::JSON_STARTING_CASH_MAP_SMALL, Config::Game::G18ZOO::JSON)
+
+      GAME_DESIGNER = 'Paolo Russo'
+
+      # Game end after the ORs in the third turn, of if any company reach 24
+      GAME_END_CHECK = { stock_market: :current_or, custom: :full_or }.freeze
+
+      OPTIONAL_RULES = [
+        # {sym: :map_b, short_name: 'Map B', desc: '5 families'}, # TODO: unblock after the map creation
+        # {sym: :map_c, short_name: 'Map C', desc: '5 families'}, # TODO: unblock after the map creation
+        # {sym: :map_d, short_name: 'Map D', desc: '7 families'}, # TODO: unblock after the map creation
+        # {sym: :map_e, short_name: 'Map E', desc: '5 families'}, # TODO: unblock after the map creation
+        # {sym: :map_f, short_name: 'Map F', desc: '5 families'}, # TODO: unblock after the map creation
+        { sym: :power_visible, short_name: 'Powers visible', desc: 'Next powers are visible since the beginning.' },
+      ].freeze
+
+      BANKRUPTCY_ALLOWED = false
+
+      HOME_TOKEN_TIMING = :float
+
+      SELL_BUY_ORDER = :sell_buy
+      SELL_AFTER = :any_time
+
+      MARKET_SHARE_LIMIT = 80 # percent
+
+      MARKET_TEXT = Base::MARKET_TEXT.merge(par_2: 'Can only enter during phase green',
+                                            par_3: 'Can only enter during phase brown').freeze
+      STOCKMARKET_COLORS = {
+        par_1: :yellow,
+        par_2: :green,
+        par_3: :brown,
+        endgame: :gray,
+      }.freeze
+
+      MUST_BUY_TRAIN = :always
+
+      # Two lays or one upgrade
+      TILE_LAYS = [{ lay: true, upgrade: true }, { lay: true, upgrade: :not_if_upgraded }].freeze
+
+      CORPORATIONS_ORDER = %w[CR GI PB PE LI TI BB EL] * 2
+
+      TILE_M7 = Tile.from_code('7 (mountain)', 'yellow', 'path=a:0,b:1;upgrade=cost:1,terrain:mountain')
```

Comment:
why are these defined like this? why don't you have them in the config?

Reply:
after you par the first corporation you will be able to par only the two corporation around; then only the next in that direction;
example;
1) par PB
2) you can par only GI or PE
3) if you par GI then you can par only CR
3) if you par PE then you can par only LI
from 4) you can par only the following (or the previous), depending on the direction you followed when you par